### ISSUE TITLE
LLVM triple overwritten issue for PTX generation

### ISSUE THREAD

**Issue by drcut** [triton-lang/triton-cpu#207]
## Bug Description
I was trying to execut Torch Inductor to generate CUDA executable code, and I got the following error:
```
'sm_89' is not a recognized processor for this target (ignoring processor)
'+ptx83' is not a recognized feature for this target (ignoring feature)
'sm_89' is not a recognized processor for this target (ignoring processor)
'sm_89' is not a recognized processor for this target (ignoring processor)
'+ptx83' is not a recognized feature for this target (ignoring feature)
'sm_89' is not a recognized processor for this target (ignoring processor)
LLVM ERROR: 64-bit code requested on a subtarget that doesn't support it!
```

I found the issue happends during LLVM->PTX transformation. The current Triton code will reset the LLVM IR triple in [createTargetMachine](https://github.com/triton-lang/triton-cpu/blob/main/python/src/llvm.cc#L61). This will set LLVM IR triple to CPU backend, which overwrites the NVIDIA triple set in [translateLLVMIRToASM](https://github.com/triton-lang/triton-cpu/blob/main/python/src/llvm.cc#L161)

## To reproduce
```
# tmp.py
# command: python tmp.py
import torch
from transformers import BertModel, BertTokenizer

model_name = "bert-base-uncased"
model = BertModel.from_pretrained(model_name)
tokenizer = BertTokenizer.from_pretrained(model_name)

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model = model.to(device)

text = "Hello, how are you today?"
inputs = tokenizer(text, return_tensors="pt").to(device)

compiled_model = torch.compile(model, backend="inductor")
# Run the compiled model
with torch.no_grad():
    compiled_outputs = compiled_model(**inputs)
print("With compile:", compiled_outputs.last_hidden_state.shape)
```

### Environment details

Triton: 8390fa2e092bfc8

**Comment by minjang** [comment]
Hi! Just curious the exact background of running into this issue. So, you're hitting a problem of generating CUDA binary in TritonCPU? I'm a bit confused. Your Torch inductor should have used Triton, not TritonCPU. What was your reason of using TritonCPU here?

**Comment by drcut** [comment]
> Hi! Just curious the exact background of running into this issue. So, you're hitting a problem of generating CUDA binary in TritonCPU? I'm a bit confused. Your Torch inductor should have used Triton, not TritonCPU. What was your reason of using TritonCPU here?

Please correct me if I am mistaken, I think the TritonCPU repo contains backend for both GPU and CPU, and if I do not set the env variable 'TRITON_CPU_BACKEND', it should get the same result as Triton for GPUs.

**Comment by minjang** [comment]
> Please correct me if I am mistaken, I think the TritonCPU repo contains backend for both GPU and CPU, and if I do not set the env variable 'TRITON_CPU_BACKEND', it should get the same result as Triton for GPUs.

This is true, but if you want to use GPU, then I'd recommend you should just use Triton (upstream) as it is the latest. TritonCPU is mostly based on a stale TritonGPU, although we periodically rebase onto the latest TritonGPU. So, my question is that you really want to use TritonCPU?

But anyhow thanks so much for reporting this bug! Yeah, without `TRITON_CPU_BACKEND`, we should have the same results of Triton's.

**Comment by drcut** [comment]
I just plan to execute the same Pytorch code on GPUs to verify the correctness of CPU backend, that's the reason I compile PTX backend on Triton-CPU.

**Comment by minjang** [comment]
Okay, I see. So you really wanted to use TritonCPU, and using the GPU for the correctness. 

Yeah, then we will take a look at this issue.

**Comment by ienkovich** [comment]
The main question here is if the problem can be reproduced using the Triton version we are currently based on. If not, then we should fix it. Otherwise, we should wait for the next rebase or file the issue in the main repo.

@drcut Do you know if the issue is reproducible with the Triton (specifically at commit 983757b3c9ccab36c9f060c85980ebd5bd5365fb)?

**Comment by minjang** [comment]
I see. I confirmed this issue. Yeah, the code was introduced for Apple macOS, but apparently overwrites NVPTX. Yes, we will fix it. Meanwhile you can comment it out.

**Comment by digantdesai** [comment]
> if I do not set the env variable 'TRITON_CPU_BACKEND', it should get the same result as Triton for GPUs.

I think this is a fair assumption. I or you can add CPU_BACKEND as an additional condition when defaulting to the host triple.

**Comment by digantdesai** [comment]
Let me know if #212 fixes this.

**Comment by minjang** [comment]
Thanks for the fix! I confirmed this fixed it on my CUDA environment.
